eventcore-types/src/projection.rs

```diff
@@ -253,6 +253,160 @@ pub trait Projector {
+///
+/// # Examples
+///
+/// ```ignore
+/// use eventcore_types::projection::MaxRetryAttempts;
+///
+/// let no_retries = MaxRetryAttempts::new(0).unwrap();
+/// let standard = MaxRetryAttempts::new(3).unwrap();
+/// let aggressive = MaxRetryAttempts::new(10).unwrap();
+///
+/// // Values above 100 are rejected
+/// assert!(MaxRetryAttempts::new(101).is_err());
+/// ```
+#[nutype(
+    validate(less_or_equal = 100),
+    derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Display)
+)]
+pub struct MaxRetryAttempts(u32);
+
+/// Backoff multiplier for exponential retry delays.
+///
+/// BackoffMultiplier represents the factor by which retry delays grow on each
+/// attempt. A value of 1.0 means constant delay (no backoff), while values
+/// greater than 1.0 implement exponential backoff.
+///
+/// The minimum value of 1.0 prevents decreasing delays, which would not make
+/// sense for retry backoff. Common values are 2.0 (double each time) or 1.5
+/// (50% increase).
+///
+/// # Examples
+///
+/// ```ignore
+/// use eventcore_types::projection::BackoffMultiplier;
+///
+/// let constant = BackoffMultiplier::new(1.0).unwrap();  // No backoff
+/// let standard = BackoffMultiplier::new(2.0).unwrap();  // Double each time
+/// let gentle = BackoffMultiplier::new(1.5).unwrap();    // 50% increase
+///
+/// // Values below 1.0 are rejected
+/// assert!(BackoffMultiplier::new(0.5).is_err());
+/// ```
+#[nutype(
+    validate(greater_or_equal = 1.0),
+    derive(Debug, Clone, Copy, PartialEq, PartialOrd, Display)
+)]
+pub struct BackoffMultiplier(f64);
+
+/// Maximum number of consecutive poll failures before stopping.
+///
+/// MaxConsecutiveFailures represents the threshold for consecutive errors
+/// during event polling. Must be at least 1 to allow for transient failures.
+///
+/// # Examples
+///
+/// ```ignore
+/// use eventcore_types::projection::MaxConsecutiveFailures;
+///
+/// let lenient = MaxConsecutiveFailures::try_new(10).unwrap();
+/// let strict = MaxConsecutiveFailures::try_new(3).unwrap();
+///
+/// // Zero failures not allowed - must allow at least one attempt
+/// assert!(MaxConsecutiveFailures::try_new(0).is_err());
+/// ```
+#[nutype(
+    validate(greater = 0, less_or_equal = 100),
+    derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Display, Into)
+)]
+pub struct MaxConsecutiveFailures(u32);
+
+/// Maximum number of retry attempts.
+///
+/// MaxRetries represents the maximum number of times to retry an operation
+/// before giving up. A value of 0 means no retries (fail immediately).
+///
+/// # Examples
+///
+/// ```ignore
+/// use eventcore_types::projection::MaxRetries;
+///
+/// let no_retry = MaxRetries::try_new(0).unwrap();
+/// let standard = MaxRetries::try_new(3).unwrap();
+/// let aggressive = MaxRetries::try_new(10).unwrap();
+/// ```
+#[nutype(
+    validate(less_or_equal = 100),
+    derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Display, Into)
+)]
+pub struct MaxRetries(u32);
```

Comment:
Again, *never* use or recommend using unwrap in production code. Also, why are we setting a completely arbitrary upper limit in library code. Leave that decision to implementers using the library.